tests/php/Service/EventServiceTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace ArchiPro\Silverstripe\EventDispatcher\Tests\Service;
+
+use ArchiPro\Silverstripe\EventDispatcher\Service\EventService;
+use ArchiPro\Silverstripe\EventDispatcher\Tests\TestListenerLoader;
+use Revolt\EventLoop;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Dev\SapphireTest;
+
+class EventServiceTest extends SapphireTest
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+    }
+
+    public function testEventDispatch(): void
+    {
+        // Create test event
+        $event = new class () {
+            public bool $handled = false;
+        };
+
+        // Create event service with real implementations
+        $service = Injector::inst()->get(EventService::class);
+
+        // Add test listener
+        $service->addListener(get_class($event), function ($event) {
+            $event->handled = true;
+        });
+
+        // Dispatch event
+        $result = $service->dispatch($event);
```

Comment:
What do you think about adding a trait to DRY this logic, I've seen it in a handful of places. Injecting in the EventService then calling dispatch/addListener.